### PR TITLE
Setup basic GA algorithm groundwork. 

### PR DESCRIPTION
**Pull request by BenDNKennedy**
Currently just maximizes an array of bits.

### CODE REVIEW

File: genetic_algorithm_logs/ga_log_20240711_153006.txt [hunk 1]
```diff
@@ -0,0 +1,3536 @@
+2024-07-11 15:30:06,587 - Initial population:
```
Comment (chyggen) [inline genetic_algorithm_logs/ga_log_20240711_153006.txt:1]: Unsure if this file should be tracked. If its mainly for debugging, then we should probably add the genetic_algorithm_logs folder to the .gitignore so it doesnt appear in our repository

File: training.py [hunk 2]
```diff
@@ -1,7 +1,67 @@
...
+    logging.info(f'Best solution fitness: {solution_fitness}')
+
+# Main execution block
 if __name__ == "__main__":
-    # do script things
-    pass
+    setup_logging()
+
+    CHROMOSOME_LENGTH = 30 
+    POPULATION_SIZE = 30
```
Comment (chyggen) [inline training.py]: Might be nice if some of the parameters we'd want to mess around with (generations, pop size) could be parsed from the command line. Forgive me if you already know this, but the argparse library is great for this and it lets you set defaults arguments for things you don't want to specify every time

**Review by chyggen — COMMENTED** [review]
Looks good! Had a few suggestions but overall really nice

File: genetic_algorithm_logs/ga_log_20240711_153006.txt (hunk 1 above)
Comment (BenDNKennedy) [reply genetic_algorithm_logs/ga_log_20240711_153006.txt:1]: Good point. I'll change that.

File: training.py (hunk 2 above)
Comment (BenDNKennedy) [reply training.py]: I agree, command line arguments are probably better for this. Will do!

File: training.py [hunk 3]
```diff
@@ -1,7 +1,76 @@
...
+
+# Main execution block
 if __name__ == "__main__":
-    # do script things
-    pass
+    setup_logging()
+
+    parser = argparse.ArgumentParser(description="Run a genetic algorithm.")
+    parser.add_argument('--chromosome_length', type=int, default=10)
+    parser.add_argument('--population_size', type=int, default=100)
```
Comment (chyggen) [inline training.py]: I hate to nitpick on this but can you add shorthand flags? Will make it less tedious to type these out down the line 
`parser.add_argument('-ps', '--population_size', type=int, default=100)`

Comment (BenDNKennedy) [reply training.py]: Sounds good, will do!

**Pull request merged**